packages/ember-data/lib/system/store/factory-locators.js

```diff
@@ -0,0 +1,74 @@
+import normalizeModelName from 'ember-data/system/normalize-model-name';
+
+function modelFor(container, modelName) {
+  var factory = modelFactoryFor(container, modelName);
+
+  if (!factory) {
+    //Support looking up mixins as base types for polymorphic relationships
+    factory = modelForMixin(container, modelName);
+  }
+  if (!factory) {
+    throw new Ember.Error("No model was found for '" + modelName + "'");
+  }
+  factory.modelName = factory.modelName || normalizeModelName(modelName);
+  factory.__container__ = factory.__container__ || container;
```

Comment:
Although this may appear drastic, this appears like the last lynch-pin preventing ED from being a good citizen in this area.